arraycontext/impl/pytato/__init__.py

```diff
@@ -296,17 +296,7 @@ def _record_leaf_ary_in_dict(key: Tuple[Any, ...],
                     raise TypeError(f"{type(self).__name__}.freeze invoked "
                                     f"with non-pytato array of type '{type(array)}'")
 
-                if subary.size == 0:
-                    # early exit for 0-sized arrays
-                    key_to_frozen_subary[key] = to_tagged_cl_array(
-                        cla.empty(self.queue.context,
-                                  shape=subary.shape,
-                                  dtype=subary.dtype,
-                                  allocator=self.allocator),
-                        get_cl_axes_from_pt_axes(subary.axes),
-                        subary.tags)
```

Comment:
It seems like this made an effort to copy the axes. How come metadata was still lost?

Reply:
Aah yep, the metadata isn't lost. I think this was just a case of misattribution.